zerver/views/auth.py

```diff
@@ -776,11 +769,6 @@ def api_fetch_api_key(request: HttpRequest, username: str=REQ(), password: str=R
         return json_error(_("Password auth is disabled in your team."),
                           data={"reason": "password auth disabled"}, status=403)
     if user_profile is None:
-        if return_data.get("valid_attestation"):
-            # We can leak that the user is unregistered iff
-            # they present a valid authentication string for the user.
-            return json_error(_("This user is not registered; do so from a browser."),
-                              data={"reason": "unregistered"}, status=403)
```

Comment:
I'm puzzled by removing this; `valid_attestation` also gets set by the social auth code path.  My guess is w only had a test for this for Google auth and need to port the test.

Reply:
But the social auth path wouldn't run in this case since the arguments in the `authenticate()` aren't valid for social auth backends, right?

Reply:
Hmm, I'll look at this again when we're back to merging this section.